nuvlaedge/orchestrator/docker.go

```diff
@@ -562,6 +563,96 @@ func (dc *DockerCoe) WaitContainerFinish(containerId string, timeout time.Durati
 	return -1, nil
 }
 
+func (dc *DockerCoe) GetContainers() ([]map[string]interface{}, error) {
+	containers, err := dc.client.ContainerList(context.Background(), container.ListOptions{All: true})
+	if err != nil {
+		return nil, err
+	}
+	var containerInfos []map[string]interface{}
+	for _, containerInfo := range containers {
+		var containerMap = make(map[string]interface{})
+		containerMap["id"] = containerInfo.ID
+		containerMap["name"] = strings.TrimPrefix(containerInfo.Names[0], "/")
+		containerMap["image"] = containerInfo.Image
+		containerMap["state"] = containerInfo.State
+		containerMap["status"] = containerInfo.Status
+		containerMap["created-at"] = time.Unix(containerInfo.Created, 0).Format(time.RFC3339)
+		containerInfos = append(containerInfos, containerMap)
+	}
+	log.Debugf("Got the Containers %v", containerInfos)
+	return containerInfos, nil
+}
+
+func (dc *DockerCoe) GetContainerStats(containerId string, statMap *map[string]interface{}) error {
```

Comment:
I don't think you need to pass the map by pointer. Maps are already a reference type so the type passed as a variable is a reference to the original type (which is also a pointer to the map structure). In this case, it would help reducing the complexity of the function a bit. Removing the need to access the content of the pointer.

Also, following the structure of the rest of the repo, I would suggest using a specific struct for it. (In that case, it would need to be a pointer). It is more difficult to maintain consistency on key names and reuse the code when "string" keys are involved.